core/include/scipp/core/dimensions.h

```diff
@@ -36,6 +37,10 @@ class SCIPP_CORE_EXPORT Dimensions {
     for (const auto [label, size] : dims)
       addInner(label, size);
   }
+  Dimensions(const std::unordered_map<Dim, scipp::index> &dims) {
+    for (const auto [label, size] : dims)
+      addInner(label, size);
+  }
```

Comment:
I do not understand this. `Dimensions` implies a particular ordering of dimensions, where `std::unordered_map` is the opposite.

Reply:
(see below)